### DISCUSSION TITLE
关于Windows下go编译exe容易误报病毒的想法。

### DISCUSSION THREAD

**Discussion by moqsien** [gvcgo/version-manager#31] [Ideas]
go语言官方的解释：https://go.dev/doc/faq#virus

1. 考虑引入CGO，通过CGO的跨平台编译方案，看看能否解决或者改善此问题。
2. osslsigncode, 见[reddit](https://www.reddit.com/r/golang/comments/s1bh01/goexecutables_and_windows_defender/)。
3. garble, 见[reddit](https://www.reddit.com/r/golang/comments/s1bh01/goexecutables_and_windows_defender/)。
4. 另一个windows免杀工具: https://github.com/awsaaaq/GoBP

觉得还是garble比较简单，youtube也有视频演示：https://www.youtube.com/watch?v=dlxqKXKMCTk
另外，gowails也在用：https://wails.io/docs/guides/obfuscated/
所以，应该无需担心会有太多不确定的问题。

**Comment by zzhaolei** [comment]
根据 reddit 上的[讨论](https://www.reddit.com/r/golang/comments/s1bh01/goexecutables_and_windows_defender/)，自签名证书似乎可以解决误报病毒的问题。

[Windows 自签名方案](https://stackoverflow.com/questions/84847/how-do-i-create-a-self-signed-certificate-for-code-signing-on-windows)

我手上没有 Windows 的机器，暂时无法验证这个方案。

**Reply by zzhaolei** [reply]
引入 cgo 应该是解决不了问题的，而且可能会导致跨平台编译时有问题，不建议引入 cgo

**Reply by moqsien** [reply]
帖子里似乎没有提到go？

**Reply by zzhaolei** [reply]
这篇帖子就是讨论的go在windows上的误报问题，帖子里也提到了在Linux上可以通过 [osslsigncode](https://github.com/mtrojnar/osslsigncode) 来为编译后的win二进制签名

**Reply by moqsien** [reply]
好吧，我没仔细看看。晚点瞅瞅。

**Reply by zzhaolei** [reply]
还有这个[混淆构建工具](https://github.com/burrowers/garble)，评论也有人说能解决误报，但是我觉得最好还是自签名的方式，签名即说明程序是正常程序

**Comment by moqsien** [comment]
windows binary的osslsigncode签名功能已经在gobuilder中实现，并应用在vmr的编译过程中。
https://github.com/gvcgo/gobuilder

@zzhaolei
